api/controllers/grades.go

```diff
@@ -0,0 +1,432 @@
+package controllers
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/UTDNebula/nebula-api/api/configs"
+	"github.com/UTDNebula/nebula-api/api/responses"
+
+	"github.com/gin-gonic/gin"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+var coursesCollection *mongo.Collection = configs.GetCollection(configs.DB, "courses")
+
+func GradesAll() gin.HandlerFunc {
+	return func(c *gin.Context) {
```

Comment:
we should look into abstracting API logic into a service layer, given the increasing complexity of the endpoints we're making.